server/src/main/java/org/apache/druid/discovery/NodeRoles.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.discovery;
+
+import com.google.common.collect.Collections2;
+import com.google.inject.Binder;
+import com.google.inject.multibindings.Multibinder;
+import org.apache.druid.guice.annotations.Global;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.server.Node;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+public class NodeRoles
+{
+  private static final Logger LOG = new Logger(NodeRoles.class);
+
+  /**
+   * Simulate the Guice binding of all node roles, but using just
+   * the known roles. Primarily for testing.
+   */
+  public static Set<NodeRole> knownRoles()
+  {
+    return new HashSet<>(Arrays.asList(NodeRole.values()));
+  }
+
+  public static void addKnownRoles(Binder binder)
+  {
+    Multibinder<NodeRole> roleBinder = binder(binder);
+    for (NodeRole role : NodeRole.values()) {
+      roleBinder.addBinding().toInstance(role);
+    }
+  }
+
+  /**
+   * Add a node role for an extension service.
+   */
+  public static void addRole(Binder binder, NodeRole role)
+  {
+    LOG.debug("Adding node role: " + role.getJsonName());
```

Comment:
Minor: it's a bit more conventional, Druid-wise, to do `log.debug("Adding node role: %s", role.getJsonName())`.